### PR TITLE
Add test cases for mizar node controller

### PR DESCRIPTION
**Pull request by Hong-Chang**
This is to add unit test for mizar node controller

**What type of PR is this?**
/kind test

### CODE REVIEW

File: pkg/controller/mizar/testutil.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+				break
+			}
+		}
+	}()
+
+	for {
+		select {
+		case <-ch:
+			return
+		case <-time.After(time.Second * 3):
```
Comment (h-w-chen) [inline pkg/controller/mizar/testutil.go:57]: generally in unit test it is not desirable to reply on timimg. Suggest to change test based on other mechanism than timer.

File: pkg/controller/mizar/util_test.go [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+	testNamespace         = "test-namespace"
+	testPhase             = "test-phase"
+	testProtocal          = "test-protocal"
+	testTenant            = "test-tenant"
+
+	testPort1 = 123
+	testPort2 = 4567
+)
+
+func TestConvertToServiceEndpointContract(t *testing.T) {
```
Comment (h-w-chen) [inline pkg/controller/mizar/util_test.go:43]: it seems multi cases i ome test. Please use the pattern of golang subtest to restructure them.

**Review by h-w-chen — APPROVED** [review]
generally looks good.
left with a few comments about minor issues. 
assuming author will take care of them properly, I provided approval.

File: pkg/controller/mizar/grpc_adaptor.go [hunk 3]
```diff
@@ -23,11 +23,29 @@ import (
...
+	DeletePod(grpcHost string, pod *v1.Pod) *ReturnCode
+	CreateNode(grpcHost string, node *v1.Node) *ReturnCode
+	UpdateNode(grpcHost string, node *v1.Node) *ReturnCode
+	DeleteNode(grpcHost string, node *v1.Node) *ReturnCode
+}
+
+type GrpcAdaptor struct {
+}
+
+// CreateArktosNetwork is to invoking grpc func of CreateArktosNetwork
```
[the remark this answers is not in the document]
Comment (Hong-Chang) [reply pkg/controller/mizar/grpc_adaptor.go:47]: Updated

File: pkg/controller/mizar/mizar_node_controller_test.go [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+	mizarNodeController.listerSynced = alwaysReady
+	go mizarNodeController.Run(mizarNodeControllerWorkerCount, ctx.Done())
+
+	go nodeInformer.Informer().Run(ctx.Done())
+	waitForMockDataReadyWithTimeout(t, grpcAdaptorMock)
+
+	testCheckEqual(t, node, grpcAdaptorMock.node)
+}
+
+func TestMizarNodeController_Update(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (Hong-Chang) [reply pkg/controller/mizar/mizar_node_controller_test.go:52]: One is to check create scenario and another is to check update scenario.  They are different in test action part. Considering the test preparing part is only 4 lines and has slight difference, I think I don't need to combine them.

File: pkg/controller/mizar/mizar_node_controller_test.go [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+	go nodeInformer.Informer().Run(ctx.Done())
+	waitForMockDataReadyWithTimeout(t, grpcAdaptorMock)
+
+	testCheckEqual(t, node, grpcAdaptorMock.node)
+	testCheckEqual(t, retryCount, grpcAdaptorMock.retryCount)
+}
+
+func syncNodeStore(nodeInformer coreinformers.NodeInformer, kubeClient *testutil.FakeNodeHandler) error {
+	list := nodeInformer.Informer().GetStore().List()
+	print(list)
```
[the remark this answers is not in the document]
Comment (Hong-Chang) [reply pkg/controller/mizar/mizar_node_controller_test.go:96]: Forgot to delete. Now deleted.

File: pkg/controller/mizar/testutil.go (hunk 1 above)
Comment (Hong-Chang) [reply pkg/controller/mizar/testutil.go:57]: After event fired, I need to verify event has been received. After waiting for 3 seconds but the event handling code hasn't been invoked, it means event has not been received. This scenario has to rely on time out.

File: pkg/controller/mizar/util_test.go (hunk 2 above)
Comment (Hong-Chang) [reply pkg/controller/mizar/util_test.go:43]: I checked and subtest is not fix for the 3 test in this file. The 3 test is not 3 different data for 1 type. It's 3 different types. The method to be tested are different.

**Pull request merged**